controllers/volumereplication_controller.go

```diff
@@ -486,6 +472,19 @@ func (r *VolumeReplicationReconciler) hasKnownGRPCError(tasks []*tasks.TaskSpec,
 	return false
 }
 
+func getReplicationState(instance *replicationv1alpha1.VolumeReplication) replicationv1alpha1.State {
```

Comment:
This will give you the requested state in the spec and won't give the actual state of the image. We should get the actual state of the image somehow and show that in the status part not the one requested.

The other way around we can update the state exact after promote/demote success that way we can show the actual state of the image.

Reply:
That's a good point.  Since not getting the actual state from the image, made below changes: 
For successful reconcile, updating the `status.state` to use `spec.ReplicationState`.  For failure scenarios, using the current `status.State` only (or `unkown` in case `status.state` is empty, say, for first reconcile).